java/src/main/java/com/google/oak/remote_attestation/InsecureAttestationVerifier.java

```diff
@@ -35,8 +62,54 @@ public class InsecureAttestationVerifier implements AttestationVerifier {
    * @return success value wrapped in a {@code Result}
    */
   @Override
-  public final Result<Boolean, Exception> verify(
-      final AttestationEvidence evidence, final AttestationEndorsement endorsement) {
-    return Result.success(true);
+  public final Result<AttestationResults, Exception> verify(long nowUtcMillis,
+      final Evidence evidence, final Endorsements endorsements,
+      final ReferenceValues referenceValues) {
+    logger.severe(evidence.getApplicationKeys().getEncryptionPublicKeyCertificate().toString());
+
+    return Result.success(
+        AttestationResults.newBuilder()
+            .setEncryptionPublicKey(com.google.protobuf.ByteString.copyFrom(extractPublicKey(
+                evidence.getApplicationKeys().getEncryptionPublicKeyCertificate().toByteArray())))
+            .build());
+  }
+
+  private static byte[] extractPublicKey(byte[] certificate) {
+    Sign1Message encryptionPublicKeyCert;
+    try {
+      encryptionPublicKeyCert = Sign1Message.deserialize(certificate);
+    } catch (CborException | CoseException e) {
+      throw new IllegalArgumentException("Failed to parse the public key certificate", e);
+    }
+
+    List<DataItem> dataItems;
+    try {
+      dataItems = CborDecoder.decode(encryptionPublicKeyCert.getMessage());
+    } catch (CborException e) {
+      throw new IllegalArgumentException(
+          "Failed to parse the encryption public key certificate payload", e);
+    }
+    for (DataItem di : dataItems) {
+      logger.severe(di.getMajorType() + ": " + di.toString());
```

Comment:
Is it expected to log everything as severe?

Reply:
Same